charmcraft/commands/store/client.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""A client to hit the Store."""
+
+import logging
+import os
+import webbrowser
+from http.cookiejar import MozillaCookieJar
+
+import appdirs
+from macaroonbakery import httpbakery
+
+from charmcraft.cmdbase import CommandError
+
+logger = logging.getLogger('charmcraft.commands.store')
+
+# XXX Facundo 2020-06-19: only staging for now; will make it "multi-server" when we have proper
+# functionality in Store's production
+BASE_URL = 'https://api.staging.snapcraft.io/publisher/api'
```

Comment:
This is definitely something that you'll want to expose as config/global option. Since it makes it easier for us doing dev to iterate. And in the future of potentially enterprise stores.

Reply:
Yes, I plan to do that when we have a second store to aim to ("production").

Reply:
Opened #51 for this